internal/postgres/toddl.go

```diff
@@ -0,0 +1,196 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package postgres
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"unicode"
+
+	"github.com/cloudspannerecosystem/harbourbridge/schema"
+	"github.com/cloudspannerecosystem/harbourbridge/spanner/ddl"
+)
+
+// schemaToDDL performs schema conversion from the source DB schema to
+// Spanner. It uses the source schema in conv.srcSchema, and writes
+// the Spanner schema to conv.spSchema.
+func schemaToDDL(conv *Conv) error {
+	for _, srcTable := range conv.srcSchema {
+		spTableName, err := GetSpannerTable(conv, srcTable.Name)
+		if err != nil {
+			return err
+		}
+		var spColNames []string
+		spColDef := make(map[string]ddl.ColumnDef)
+		conv.issues[srcTable.Name] = make(map[string][]schemaIssue)
+		// Iterate over columns using ColNames order.
+		for _, srcColName := range srcTable.ColNames {
+			srcCol := srcTable.ColDef[srcColName]
+			colName, err := GetSpannerCol(conv, srcTable.Name, srcCol.Name, false)
+			spColNames = append(spColNames, colName)
+			if err != nil {
+				return fmt.Errorf("can't get Spanner col: %w", err)
+			}
+			ty, issues := toSpannerType(conv, srcCol.Type.Id, srcCol.Type.Mods)
+			if len(srcCol.Type.ArrayBounds) > 1 {
+				ty = ddl.String{ddl.MaxLength{}}
+				issues = append(issues, multiDimensionalArray)
+			}
+			// TODO: add issues for all elements of srcCol.Ignored.
+			if srcCol.Ignored.ForeignKey {
+				issues = append(issues, foreignKey)
+			}
+			if srcCol.Ignored.Default {
+				issues = append(issues, defaultValue)
+			}
+			if len(issues) > 0 {
+				conv.issues[srcTable.Name][srcCol.Name] = issues
+			}
+			spColDef[colName] = ddl.ColumnDef{
+				Name:    colName,
+				T:       ty,
+				IsArray: len(srcCol.Type.ArrayBounds) == 1,
+				NotNull: srcCol.NotNull,
+				Comment: "From: " + quoteIfNeeded(srcCol.Name) + " " + prType(srcCol.Type),
```

Comment:
nits: `prettifyTypeFormat` or `prettifyType`?

Reply:
Done.
Changed to printSourceType.